terraform-provider/README.md

```diff
@@ -0,0 +1,86 @@
+# Cosmo Terraform Provider
+
+This repository is for the [Cosmo](https://registry.terraform.io/wundergraph/cosmo) Terraform provider, designed to manage Cosmo resources within Terraform. It includes a resource and a data source, examples, and generated documentation.
+
+## Requirements
+
+- [Terraform](https://developer.hashiCorp.com/terraform/downloads) >= 1.0
+- [Go](https://golang.org/doc/install) >= 1.21
+
+## Implemented Resources
+
+The Cosmo Terraform provider includes the following resources and data sources:
+
+### Resources
+
+- [cosmo_namespace](docs/resources/namespace.md): Manages namespaces within Cosmo.
+- [cosmo_monograph](docs/resources/monograph.md): Manages monographs in Cosmo.
+- [cosmo_federated_graph](docs/resources/federated_graph.md): Manages federated graphs in Cosmo.
+- [cosmo_subgraph](docs/resources/subgraph.md): Manages subgraphs in Cosmo.
+
+### Data Sources
+
+- [cosmo_namespace](docs/data-sources/namespace.md): Retrieves information about namespaces in Cosmo.
+- [cosmo_monograph](docs/data-sources/monograph.md): Retrieves information about monographs in Cosmo.
+- [cosmo_federated_graph](docs/data-sources/federated_graph.md): Retrieves information about federated graphs in Cosmo.
+- [cosmo_subgraph](docs/data-sources/subgraph.md): Retrieves information about subgraphs in Cosmo.
+
+Each resource and data source allows you to define and manage specific aspects of your Cosmo infrastructure seamlessly within Terraform.
+
+## Building The Provider
+
+To build the provider, clone the repository, enter the directory, and run `make install` to compile and install the provider binary. Note that the `install` command will first build the provider to ensure the binary is up to date.
+
+## Usage
```

Comment:
We should add a simple example in the README or link to an example prominently.

Reply:
checkout: https://github.com/wundergraph/terraform-provider-cosmo/pull/1